pinot-broker/src/main/java/org/apache/pinot/broker/failuredetector/BaseExponentialBackoffRetryFailureDetector.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.broker.failuredetector;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.concurrent.ThreadSafe;
+import org.apache.pinot.common.metrics.BrokerGauge;
+import org.apache.pinot.common.metrics.BrokerMetrics;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.utils.CommonConstants.Broker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The {@code BaseExponentialBackoffRetryFailureDetector} is a base failure detector implementation that retries the
+ * unhealthy servers with exponential increasing delays.
+ */
+@ThreadSafe
+public abstract class BaseExponentialBackoffRetryFailureDetector implements FailureDetector {
+  protected final Logger _logger = LoggerFactory.getLogger(getClass());
```

Comment:
I'm curious why this (and the `_name` variable) aren't static. Is there a convention in the Pinot code base? Just FYI, I found 437 instances of `static final Logger LOGGER`, and 14 of `final Logger _logger` (a few of which were actually static).

Reply:
This is a base class, and we want to log the actual class name that extends this base class, so it cannot be static. E.g. I want the logger to log under `ConnectionFailureDetector` instead of `BaseExponentialBackoffRetryFailureDetector`

Reply:
Thanks for clarifying...but why do you want classes that extend `BaseExponentialBackoffRetryFailureDetector` to use its logger? Is there some issue with the concrete class having its own logger?

Just asking because I see 25+ instances of abstract classes with private static loggers, but only one other class `BaseSegmentJob` that does the same thing as here.